### PR TITLE
To next js

### PR DESCRIPTION
**Pull request by hunterwilliams**
Items left to do

- [x] *Hunter* Ensure build works for CI & github
- ~[ ] *Hunter* Fix overall images~
- [x] *Hunter* Fix google tag manager; after remove index.html
- [x] *Hunter* Ensure linking hash in ourwork page - **swapped to use query param**
- [x] *Hunter* decide whether to **commit** or omit .mjs files
- [x] *Vl3oss* Make carousel work again - fixing shaders loader

### CODE REVIEW

**Comment by hunterwilliams** [comment]
Carousel is quite stuck due to NextJS dynamic not working. Not sure why.

It uses "document" references therefore must use Next/Dynamic. However had no luck in implementing it. Tried both rescript use of dynamic and importing it via straight js in the next page. Even simpler components no dice. Could be nextjs being v11 or my node version tried both v14 and v15.

Issues come across:
* Import getting transpiled to require -> turning experimental feature on nextjs causes it to say element is invalid
* If working it says invalid element or div issue
* Saying it's an object and not something that can be rendered or maybe it thinks it's an array

Basically hit all of the issues of this... but not able to find a solution https://forum.rescript-lang.org/t/next-js-dynamic-imports-server-and-client-markup-mismatch/1652/9

**Comment by Vl3oss** [comment]
@hunterwilliams I added a working carousel version with this commit 1354a78cb38306b644146c1ab0d6c3a8e20d2ecb .

I'm not sure why when use dynamic import, it tries to use commonjs instead of esmodule. My guess is probably babel?
So, to fix that I turn on Next's experimental feature that make esmodule work, 
```
experimental.esmExternals: 'loose'
```
from [here](https://nextjs.org/docs/messages/import-esm-externals)

It seems to be working now, but currently in raw js. Maybe will clean it up a bit later.

**Comment by hunterwilliams** [comment]
Wonky but it works. Thanks @Vl3oss

**Comment by hunterwilliams** [comment]
It's now working and tested in MAMP. Should be fine for launch. Incoming redesign and content changes, but want to get it merged before that as this PR is big enough
